tanner/api/api.py

```diff
@@ -32,8 +32,8 @@ def __init__(self, redis_client):
                                       'cmd_exec': 0}
 
         for sess in sessions:
-            result['total_duration'] += sess['end_time'] - sess['start_time']
-            for attack in sess['attack_types']:
+            result['total_duration'] += sessions[sess]['end_time'] - sessions[sess]['start_time']
```

Comment:
First, please submit any bug fix in a separate PR.
I don't understand this change, `sessions` is of type `list`, right?

Reply:
Its just a minor change. `sessions` is of type `dictionary`.

Reply:
I think [this](https://github.com/mushorg/tanner/blob/d55bf19b06d1430c3846896f4dfd8e1b3ce06c14/tanner/api/api.py#L53) will return a `list`. Also, I checked and it's a `list`

Reply:
Yup! it returns a list with the whole string `'{"sess1": {"end_time": 2.00, "start_time": 0.00 }, "sess2": {"attack_types": ["rfi"]}}'` at `0th` position. Its bcoz I made `Asyncmock` return only one element in `query_res` just to make it a bit simple.

Reply:
I'm not sure why are you changing code to make test cases.

Reply:
Any updates here?

Reply:
Everything looks good but I'm still not convinced here.

Reply:
`sessions` is a dictionary. So, iterating over it means iterating the keys which is `string` so it raises `TypeError: string indices must be integers.`  This is just a workaround for it.

https://stackoverflow.com/questions/49695836/typeerror-string-indices-must-be-integers-python

Reply:
No, I'm saying this is a list object.
Look here, it should return `list` object. And I have checked on my system the previous code is running fine. So I don't understand why you are calling it a dict object.

Reply:
I guess its not very much clear from the code. Can you give a sample `session` object if it has to be a list?

Reply:
Quick question first, did you change the code because you get an error while running or just by intuition that `sessions` is a `dict` object?

Reply:
It was based on intution. I was having a doubt if its a list because of `json.loads`.

Reply:
I think you should run `tannerweb` and then go to `localhost:8091/snare-stats/<snare-uuid>` which calls that function and see if it is actually properly (maybe analyze the variables types).

Reply:
Roger! Ill analyze this more deeply.